e2e/client.test.js

```diff
@@ -0,0 +1,451 @@
+    }
+    update: {
+      name: "${process.env.SERVICE_NAME}"
+      algoliaAppId: "${process.env.ALGOLIA_APP_ID}"
+      algoliaApiKey: "${process.env.ALGOLIA_API_KEY_ALL}"
+      auth0Domain: "${process.env.AUTH0_DOMAIN}"
+      auth0ClientId: "${process.env.AUTH0_CLIENT_ID}"
+    }
+  ) {
+    id
+    name
+    title
+    auth0Domain
+    auth0ClientId
+    authorizedDomains
+    algoliaAppId
+    algoliaApiKey
+    algoliaSynonyms
+    mailgunDomain
+    mailgunApiKey
+    slackChannelHook
+    tagCategories {
+      order
+      name
+      labels {
+        id
+        order
+        name
+      }
+    }
+    workplaceSharing
+    bugReporting
+  }
+}`
+
+const upsertConfigMutation = async apiContext => {
+  const res = await apiContext.post('/', {
+    data: {
+      query: upsertConfig
+    }
+  })
+  const jsonRes = await res.json()
+  const results = await jsonRes.data
+  return results
+}
+
+const tagsId = `query GetAllTags{
+  tagLabels {
+    id
+    name
+  }
+}`
+
+const tagsIdQuery = async apiContext => {
+  const res = await apiContext.post('/', {
+    data: {
+      query: tagsId
+    }
+  })
+  const jsonRes = await res.json()
+  const rawResults = await jsonRes.data.tagLabels
+  const results = await rawResults.reduce((accumulator, current) => {
+    if (!accumulator.find(item => item.name === current.name)) {
+      accumulator.push(current)
+    }
+    return accumulator
+  }, [])
+  const randomNumber = Math.floor(Math.random() * results.length)
+  let randomAddNumber = Math.floor(Math.random() * results.length)
+  do {
+    randomAddNumber = Math.floor(Math.random() * results.length)
+  } while (randomNumber === randomAddNumber)
+  const tagLabel = await results[randomNumber]
+  const tagAddLabel = await results[randomAddNumber]
+  const { id: tagId, name: tagName } = await tagLabel
+  const { id: tagAddId, name: tagAddName } = await tagAddLabel
+  return { tagId, tagName, tagAddId, tagAddName }
+}
+
+const deleteAll = field => {
+  return `mutation DeleteAll{
+      ${field} {
+        count
+      }
+    }`
+}
+
+const questionsText = [
+  'Ceci est une question',
+  "Message de 2023 au futur : le réchauffement climatique c'est vraiment nul",
+  "En fait, il n'y a aucune question",
+  '𓇋𓅱𓄙𓅓 ... bonne chance pour faire fonctionner la recherche avec ça',
+  'Lorem ipsum ...',
+  'MMMMDCCLXXXVIII - MMMMDCLXXXVIII',
+  'Vous êtes libre de changer les textes des questions'
+]
+
+const answersText = [
+  "Message du futur à 2023 : au secours c'est encore pire",
+  'Ceci est une réponse',
+  'C',
+  '𓁷𓏤𓎟𓀀𓁐𓏥𓃀𓈖𓌱𓅓𓎛𓅱𓀔𓈖𓌱𓅓𓎛𓇋𓇋𓏏𓁐𓐍𓂋𓋴𓂝𓎛𓋩𓉔𓊪𓏛𓋴𓐠𓄿𓂋𓏏𓌗𓀁𓌷𓂝𓏏𓏭𓏛𓇾𓏏𓅓𓅱𓀀𓁐𓏪𓃀𓌢𓌢𓈖𓈖𓏛',
+  'Vous avez aussi le droit de modifier le texte des réponses'
+]
+
+const uniqueRandom = (obj, ...compareNumbers) => {
+  let uniqueNumber
+  do {
+    uniqueNumber = Math.floor(Math.random() * obj.length)
+  } while (compareNumbers.includes(uniqueNumber))
+  return uniqueNumber
+}
```

Comment:
Je suggère quelques renommages
```suggestion
const randomIndex = (array, ...exclusions) => {
  let result
  do {
    result = Math.floor(Math.random() * array.length)
  } while (exclusions.includes(result))
  return result
}
```

Reply:
Quand la partie random sera retirée, ceci ne sera plus pertinent.